superset/dashboards/filters.py

```diff
@@ -107,9 +119,14 @@ def apply(self, query: Query, value: Any) -> Query:
         query = query.filter(
             or_(
                 Dashboard.id.in_(owner_ids_query),
-                Dashboard.id.in_(published_dash_query),
+                Dashboard.id.in_(datesource_perm_query),
                 Dashboard.id.in_(users_favorite_dash_query),
+                Dashboard.id.in_(roles_based_query),
             )
         )
 
         return query
+
+    @staticmethod
+    def has_no_role_based_access() -> bool:
```

Comment:
This could probably also be changed to an inline variable as opposed to a static method.

In general, unless a function needs to me exposed as a convenient public API, I'd recommend avoid using static methods, so to keep things local.